core/services/job/orm.go

```diff
@@ -924,20 +924,19 @@ func (o *orm) FindJobsByPipelineSpecIDs(ids []int32) ([]Job, error) {
 func (o *orm) PipelineRuns(jobID *int32, offset, size int) (runs []pipeline.Run, count int, err error) {
 	err = o.q.Transaction(func(tx pg.Queryer) error {
 		var args []interface{}
-		var where string
+		var filter string
 		if jobID != nil {
-			where = " WHERE jobs.id = $1"
+			filter = "JOIN jobs USING(pipeline_spec_id) WHERE jobs.id = $1" // TODO:  add support for more than 1 jobID?
```

Comment:
I added this TODO note, because I noticed that we are parsing jobID into a list of jobIDs (args), but then we only use the first jobID and discard the rest.